### ISSUE TITLE
Invalid SQL query in 3.1.2

### ISSUE THREAD

**Issue by lrtherond** [mautic/mautic#9377]
The error shows up in the logs as follows:

```
[2020-11-07 19:02:55] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\SyntaxErrorException: "An exception occurred while executing 'SELECT count(DISTINCT .ph.lead_id) as click_count, cs.leadlist_id FROM channel_url_trackables cut INNER JOIN page_hits ph ON ph.redirect_id = cut.redirect_id AND ph.source = cut.channel AND ph.source_id = cut.channel_id INNER JOIN lead_lists_leads cs ON cs.lead_id = ph.lead_id WHERE (cut.channel = ?) AND (cut.channel_id IN (2)) AND (cs.leadlist_id IN (2)) AND (ph.date_hit BETWEEN ? AND ?) GROUP BY cs.leadlist_id' with params ["email", "2020-10-07 00:00:00", "2020-11-07 19:02:55"]:  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '.ph.lead_id) as click_count, cs.leadlist_id FROM channel_url_trackables cut INNE' at line 1" at /var/www/luxuriant.software/html/MAU/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 79 {"exception":"[object] (Doctrine\\DBAL\\Exception\\SyntaxErrorException(code: 0): An exception occurred while executing 'SELECT count(DISTINCT .ph.lead_id) as click_count, cs.leadlist_id FROM channel_url_trackables cut INNER JOIN page_hits ph ON ph.redirect_id = cut.redirect_id AND ph.source = cut.channel AND ph.source_id = cut.channel_id INNER JOIN lead_lists_leads cs ON cs.lead_id = ph.lead_id WHERE (cut.channel = ?) AND (cut.channel_id IN (2)) AND (cs.leadlist_id IN (2)) AND (ph.date_hit BETWEEN ? AND ?) GROUP BY cs.leadlist_id' with params [\"email\", \"2020-10-07 00:00:00\", \"2020-11-07 19:02:55\"]:\n\nSQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '.ph.lead_id) as click_count, cs.leadlist_id FROM channel_url_trackables cut INNE' at line 1 at /var/www/luxuriant.software/html/MAU/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:79, Doctrine\\DBAL\\Driver\\PDOException(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '.ph.lead_id) as click_count, cs.leadlist_id FROM channel_url_trackables cut INNE' at line 1 at /var/www/luxuriant.software/html/MAU/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:123, PDOException(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '.ph.lead_id) as click_count, cs.leadlist_id FROM channel_url_trackables cut INNE' at line 1 at /var/www/luxuriant.software/html/MAU/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:121)"} []
```

When properly formatted, the questionable SQL query looks like this:

```
SELECT
  count(DISTINCT.ph.lead_id) as click_count,
  cs.leadlist_id
FROM
  channel_url_trackables cut
  INNER JOIN page_hits ph ON ph.redirect_id = cut.redirect_id
  AND ph.source = cut.channel
  AND ph.source_id = cut.channel_id
  INNER JOIN lead_lists_leads cs ON cs.lead_id = ph.lead_id
WHERE
  (cut.channel = ?)
  AND (cut.channel_id IN (2))
  AND (cs.leadlist_id IN (2))
  AND (
    ph.date_hit BETWEEN ?
    AND ?
  )
GROUP BY
  cs.leadlist_id ' with params ["email", "2020-10-07 00:00:00", "2020-11-07 18:55:26"]
```

Notice the following:

```
  (cut.channel = ?)
```

This happens when accessing an email under Channels > Emails.

**Comment by lrtherond** [comment]
[
<img width="2350" alt="Screen Shot 2020-11-07 at 3 57 59 PM" src="https://user-images.githubusercontent.com/1578548/98453038-22171c80-2112-11eb-8e5d-6b0e2927e488.png">
](url)

Maybe that talks to someone

**Comment by lrtherond** [comment]
https://user-images.githubusercontent.com/1578548/98453038-22171c80-2112-11eb-8e5d-6b0e2927e488.png

**Comment by mreall** [comment]
It looks like there's an extra period before "ph.lead_id" in the SELECT statement:
```
SELECT
  count(DISTINCT .ph.lead_id)
-----------------^
```

When I remove that the query runs fine.

**Comment by mreall** [comment]
I removed the extra period from lines 752 and 784 in app/bundles/EmailBundle/Model/EmailModel.php and the email chart now loads fine.

**Comment by npracht** [comment]
@mreall could you please create a PR with that fix ?
Thanks !

**Comment by mreall** [comment]
@npracht I created a PR. The contributor agreement I submitted will need to be approved first.

**Comment by RCheesley** [comment]
> The contributor agreement I submitted will need to be approved first.

Done! ✅

**Comment by npracht** [comment]
Congratulation !

**Comment by npracht** [comment]
Solved in 3.2.0
